File: mycollab-web/src/main/java/com/esofthead/mycollab/module/project/view/settings/SettingUrlResolver.java
package com.esofthead.mycollab.module.project.view.settings;

import com.esofthead.mycollab.common.UrlEncodeDecoder;
import com.esofthead.mycollab.module.project.events.ProjectEvent;
import com.esofthead.mycollab.module.project.view.ProjectUrlResolver;
import com.esofthead.mycollab.module.project.view.parameters.ProjectScreenData;
import com.esofthead.mycollab.module.project.view.parameters.ProjectSettingScreenData;
import com.esofthead.mycollab.vaadin.events.EventBus;
import com.esofthead.mycollab.vaadin.mvp.PageActionChain;

public class SettingUrlResolver extends ProjectUrlResolver {
	public SettingUrlResolver() {
		this.addSubResolver("notification", new ViewUrlResolver());
	}

	private static class ViewUrlResolver extends ProjectUrlResolver {
		@Override
		protected void handlePage(String... params) {
			String decodeUrl = UrlEncodeDecoder.decode(params[0]);
			int projectId = Integer.parseInt(decodeUrl);

			PageActionChain chain = new PageActionChain(
					new ProjectScreenData.Goto(projectId),
					new ProjectSettingScreenData.ViewNotification());
			EventBus.getInstance().fireEvent(
					new ProjectEvent.GotoMyProject(this, chain));
		}
	}
}

File: mycollab-web/src/main/scala/com/esofthead/mycollab/module/project/view/settings/SettingUrlResolver.scala
package com.esofthead.mycollab.module.project.view.settings

import com.esofthead.mycollab.common.UrlTokenizer
import com.esofthead.mycollab.eventmanager.EventBusFactory
import com.esofthead.mycollab.module.project.events.ProjectEvent
import com.esofthead.mycollab.module.project.view.ProjectUrlResolver
import com.esofthead.mycollab.module.project.view.parameters.{ProjectScreenData, ProjectSettingScreenData}
import com.esofthead.mycollab.vaadin.mvp.PageActionChain

/**
  * @author MyCollab Ltd
  * @since 5.0.9
  */
class SettingUrlResolver extends ProjectUrlResolver {
  protected override def handlePage(params: String*) {
    val projectId = new UrlTokenizer(params(0)).getInt
    val chain = new PageActionChain(new ProjectScreenData.Goto(projectId),
      new ProjectSettingScreenData.ViewSettings)
    EventBusFactory.getInstance.post(new ProjectEvent.GotoMyProject(this, chain))
  }
}
